nbs/15_gsa_gls/07-compile_gwas_snps_and_twas_genes.ipynb

```diff
@@ -0,0 +1,2355 @@
+    "    / REFERENCE_PANEL.lower()\n",
+    "    / EQTL_MODEL.lower()\n",
+    ")\n",
+    "\n",
+    "OUTPUT_DIR_BASE.mkdir(parents=True, exist_ok=True)\n",
+    "\n",
+    "display(f\"Using output dir base: {OUTPUT_DIR_BASE}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b6ebb422-8204-494b-8c72-360a71ce9863",
+   "metadata": {
+    "papermill": {
+     "duration": 0.016156,
+     "end_time": "2021-04-28T22:27:17.704727",
+     "exception": false,
+     "start_time": "2021-04-28T22:27:17.688571",
+     "status": "completed"
+    },
+    "tags": []
+   },
+   "source": [
+    "# Load MultiPLIER Z genes"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c48ef527",
+   "metadata": {
+    "papermill": {
+     "duration": 0.050172,
+     "end_time": "2021-04-28T22:27:17.770996",
+     "exception": false,
+     "start_time": "2021-04-28T22:27:17.720824",
+     "status": "completed"
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "multiplier_z_genes = pd.read_pickle(\n",
+    "    conf.MULTIPLIER[\"MODEL_Z_MATRIX_FILE\"]\n",
+    ").index.tolist()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0a4ecbd9",
+   "metadata": {
+    "papermill": {
+     "duration": 0.027596,
+     "end_time": "2021-04-28T22:27:17.819065",
+     "exception": false,
+     "start_time": "2021-04-28T22:27:17.791469",
+     "status": "completed"
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "len(multiplier_z_genes)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a18ce359-0759-4260-ae6f-053e8c0041b7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "assert len(multiplier_z_genes) == len(set(multiplier_z_genes))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "115e8864",
+   "metadata": {
+    "papermill": {
+     "duration": 0.027396,
+     "end_time": "2021-04-28T22:27:17.863147",
+     "exception": false,
+     "start_time": "2021-04-28T22:27:17.835751",
+     "status": "completed"
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "multiplier_z_genes[:5]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7bed0ce0-e7c7-4300-9a61-5412450b1904",
+   "metadata": {
+    "papermill": {
+     "duration": 0.006254,
+     "end_time": "2022-07-11T00:59:29.336579",
+     "exception": false,
+     "start_time": "2022-07-11T00:59:29.330325",
+     "status": "completed"
+    },
+    "tags": []
+   },
+   "source": [
+    "# GWAS"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "331e5691-c014-43e4-b94c-060c64d329df",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "gwas_file_columns = pd.read_csv(GWAS_FILE, sep=\"\\t\", nrows=2).columns\n",
+    "\n",
+    "assert (\n",
+    "    \"panel_variant_id\" in gwas_file_columns\n",
+    "), \"GWAS file must be the final imputed one with column 'panel_variant_id'\"\n",
```

Comment:
What does this mean, "the final imputed one"?

Reply:
It means the final imputed GWAS. I updated the description.